### ISSUE TITLE
Developer testing mode

### ISSUE THREAD

**Issue by zmughal** [PDLPorters/pdl#10]
Don't overwrite code gen files if they are the same so that `make` will not have to rebuild frequently. Simple sketch of the flow:

``` shell
codegen $FILE.in > $FILE.new; md5sum $FILE $FILE.new; mv $FILE.new $FILE
```

Also, make sure all the files loaded during test are from the build directory rather than a pre-existing PDL install. Use `strace` to verify this.

**Comment by mohawk2** [comment]
That's a good idea! I think checkout %INC might be an easier way than strace.

**Comment by zmughal** [comment]
OK, to clarify this more, if I have a `.pd` file that generates both native code and Perl code, I should not have to wait for things to compile if I only change the Perl part in places that don't change any of the compiled code. I'm using Perl for a reason!

**Comment by mohawk2** [comment]
When I converted PDL::Ops (I think) to use Inline::Pdlpp, there's a PMCode param that triggers making a `_*_int` method name to be called by the included Perl. My initial look at PDL::PP didn't show how to trigger that without still leaving in the PMCode, even though that wasn't directly being used. I was then able to avoid recompiles where I only changed the Perl code / docs. What I'm saying is, this is a thing I have working.
